sched/sched/sched_timerexpiration.c

```diff
@@ -238,7 +238,18 @@ static uint32_t nxsched_cpu_scheduler(int cpu, uint32_t ticks,
     {
       /* Apply the keep alive hack */
 
-      return KEEP_ALIVE_TICKS;
+      /* The idle task always lies at the end of the task list,
+       * idle task no need time slice.
+       */
+
+      if (ntcb->flink != NULL)
+        {
+          return KEEP_ALIVE_TICKS;
+        }
+      else
+        {
+          return UINT32_MAX - 1;
```

Comment:
I think we can remove else clause, the caller could handle zero return value as maximum timeout correctly.

Reply:
@mkeydevelop could you try this patch: https://github.com/apache/incubator-nuttx/pull/6746

Reply:
It work, but up_alarm_start/up_timer_start never call if only idle task is runnable